src/cli/commands/test/iac-local-execution/index.ts

```diff
@@ -88,17 +88,23 @@ export async function test(
     // we may have loaded and parsed all but terraform files in the previous step
     // so now we check if we need to do a second load and parse which dereferences TF vars
     if (validFileTypes && !validFileTypes.includes(ValidFileType.Terraform)) {
-      const {
-        allParsedFiles,
-        allFailedFiles,
-      } = await loadAndParseTerraformFiles(
-        pathToScan,
-        parsedFiles,
-        failedFiles,
-        options.detectionDepth,
-      );
-      parsedFiles = allParsedFiles;
-      failedFiles = allFailedFiles;
+      try {
+        const {
+          parsedFiles: tfParsedFiles,
+          failedFiles: tfFailedFiles,
+        } = await loadAndParseTerraformFiles(
+          pathToScan,
+          options.detectionDepth,
+        );
+        parsedFiles = parsedFiles.concat(tfParsedFiles);
+        failedFiles = failedFiles.concat(tfFailedFiles);
+      } catch (err) {
+        if (parsedFiles.length > 0 && err instanceof NoFilesToScanError) {
+          // ignore this error since we might not have .tf files in the folder and we have separated them
+        } else {
+          throw err;
+        }
```

Comment:
Teo and I discussed this in person, this caught me out initially as I missed that we're relying on `parsedFiles` here to be the length of the array set on line 75 rather than on 99. This logic perhaps indicates the need to refactor this flow in the near future to redesign the flow so there is a single entry point here to a function that rethinks the file discovery and parsing to encapsulate the need for multifile scans.

In the short term changing the comment to something like:
```suggestion
        if (parsedFiles.length > 0 && err instanceof NoFilesToScanError) {
          // ignore this error since we've discovered non-terraform files in the previous block
        } else {
          throw err;
        }
```